lib/ui/painting.dart

```diff
@@ -552,20 +615,26 @@ class Paint {
         result.write(' hairline');
       if (strokeCap != StrokeCap.butt)
         result.write(' $strokeCap');
+      if (strokeJoin == StrokeJoin.miter) {
+        if (strokeMiterLimit != _kStrokeMiterLimitDefault)
+          result.write('$strokeJoin up to $strokeMiterLimit');
```

Comment:
While we're here (and sorry to make you change this oven and over), can we actually make a slight tweak to this?
Printing doubles ends up with ugly strings of digits. What we normally do is print them to 1dp.

This would mean changing the strokeWidth case above to:
```dart
         result.write(' ${strokeWidth.toStringAsFixed(1)}');
```
...and this case to:
```dart
          result.write('$strokeJoin up to ${strokeMiterLimit.toStringAsFixed(1)}');
```